src/main/scala/akka/persistence/jdbc/query/scaladsl/JdbcReadJournal.scala

```diff
@@ -139,46 +141,87 @@ class JdbcReadJournal(config: Config)(implicit val system: ExtendedActorSystem)
     else {
       readJournalDao.eventsByTag(tag, offset, latestOrdering.maxOrdering, max)
         .mapAsync(1)(Future.fromTry)
-        .mapConcat { case (repr, _, row) =>
-          adaptEvents(repr).map(r => EventEnvelope(Sequence(row.ordering), r.persistenceId, r.sequenceNr, r.payload))
+        .mapConcat {
+          case (repr, _, row) =>
+            adaptEvents(repr).map(r => EventEnvelope(Sequence(row.ordering), r.persistenceId, r.sequenceNr, r.payload))
         }
     }
   }
 
   /**
-    * @param terminateAfterOffset If None, the stream never completes. If a Some, then the stream will complete once a
-    *                             query has been executed which might return an event with this offset (or a higher offset).
-    *                             The stream may include offsets higher than the value in terminateAfterOffset, since the last batch
-    *                             will be returned completely.
-    */
+   * @param terminateAfterOffset If None, the stream never completes. If a Some, then the stream will complete once a
+   *                             query has been executed which might return an event with this offset (or a higher offset).
+   *                             The stream may include offsets higher than the value in terminateAfterOffset, since the last batch
+   *                             will be returned completely.
+   */
   private def eventsByTag(tag: String, offset: Long, terminateAfterOffset: Option[Long]): Source[EventEnvelope, NotUsed] = {
+
     import akka.pattern.ask
     implicit val askTimeout: Timeout = Timeout(readJournalConfig.journalSequenceRetrievalConfiguration.askTimeout)
-    /* We unfold with 3 parameters:
-     * - The first parameter (from) determines the starting offset (i.e. the minimum value) to query from.
-     * - The second parameter (completeImmediately) signifies that we have queried events until at least
-     *   `terminateAfterOffset`, therefore the stream no longer needs to continue.
-     * - The third parameter (retrieveImmediately) is a boolean which determines whether the database query should be
-     *   executed immediately, or after a delay.
+
+    sealed trait FlowControl
```

Comment:
To make this method a bit shorter, we could probably move this sealed trait to the compation object of this JdbcReadJournal (then we should probably also make it private since it it not meant to be used outside)